eos/meson-mixing/bq-mixing.hh

```diff
@@ -0,0 +1,49 @@
+/* vim: set sw=4 sts=4 et foldmethod=syntax : */
+
+/*
+ * Copyright (c) 2021 Danny van Dyk
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef EOS_GUARD_EOS_MESON_MIXING_BQ_MIXING_HH
+#define EOS_GUARD_EOS_MESON_MIXING_BQ_MIXING_HH 1
+
+#include <eos/utils/complex.hh>
+#include <eos/utils/options.hh>
+#include <eos/utils/parameters.hh>
+#include <eos/utils/private_implementation_pattern.hh>
+#include <eos/utils/reference-name.hh>
+
+namespace eos
+{
+    class BMixing :
+        public ParameterUser,
+        public PrivateImplementationPattern<BMixing>
+    {
+        public:
+            BMixing(const Parameters & parameters, const Options & options);
+            ~BMixing();
+
+            // Observables
```

Comment:
I wonder whether also M_12 should be an "observable". Then it can be combined in the future with observables in other decays, i.e. for example all the time-dependent CP asymmetries in all kind of B-> M_1 M_2 decays, which allow for mixing-induced CP violation etc.

Reply:
Meril and I think that M_12 is not an observable. To access the phase or any other aspect of M_12, we can simply use the BMixing class in any other class.

Reply:
Even when it is hidden by the "Implementationpattern"?